website/cue/reference/components.cue

```diff
@@ -439,11 +443,26 @@ components: {
 		default_namespace: string
 	}
 
-	#Output: {
+	#OutputData: {
 		logs?:    #LogOutput
 		metrics?: #MetricOutput
 	}
 
+	#Output: {
+		name:        string
+		description: string
+	}
+
+	#Outputs: {
+		_default_output: #Output & {
+			name:        ""
+			description: "Default output stream of the component. Use this component's ID as an input to downstream transforms and sinks."
+		}
+
+		default_output: *_default_output | #Output | null
+		named_outputs: [...#Output]
```

Comment:
Similar to above, it feels like it'd be a little safer to adjust `named_outputs` to be `outputs` and then make it an enforced either/or situation with `default_output`. That way there's less chance of us forgetting to put `default_ouput: null` in cases where we've already filled out `outputs`.

Reply:
Hmm, in that case, it seems like we'd have to explicitly include `_default_output` in the `outputs` list for any component that uses multiple outputs and a default. In the current version, `default_output: null` lets components without a default explicitly leave it off. I'm happy to change it, but I feel like the current version might be a bit easier. Does that make sense?

Reply:
Ah, I see what you mean. I guess we don't have a great idea yet what will be the more common pattern. In the code we opted to require the default output in the `fn outputs(&self) -> Vec<Output>` return value, so there's at least some benefit to being consistent. As far as existing use cases, it seems like we have:

1. Remap with a primary default and an optional `dropped` output
2. Route with no default and all named outputs
3. Datadog agent (as of #10776) with either one default or two named outputs and no default
4. Everything else with just a default output

That makes it seem like the default output isn't so common in components with multiple outputs that we should always assume it's there. So I think I'd still lean towards requiring the default to be listed in cases where we're specifying multiple outputs, but it's hard to know for sure. I'm curious if @jszwedko has an opinion here.

Reply:
I'm 👍 on the current modeling of `#Outputs: *[_default_output] | [#Output, ...#Output]`. That is that it defaults to just a default output and, if you add outputs, you can include `_default_output`. I agree with Luke that, if you are editing `outputs` you are more likely to think to add `_default_output` when needed than remember to set `default_output` to false to override.